databricks/koalas/indexes.py

```diff
@@ -1540,6 +1548,75 @@ def argmin(self):
 
         return sdf.orderBy(self.spark.column.asc(), F.col(sequence_col).asc()).first()[0]
 
+    def putmask(self, mask, value):
+        """
+        Return a new Index of the values set with the mask.
+
+        .. note:: this API can be pretty expensive since it is based on
+             a global sequence internally.
+
+        Returns
+        -------
+        Index
+
+        Example
+        -------
+        >>> kidx = ks.Index([1, 2, 3, 4, 5])
+        >>> kidx
+        Int64Index([1, 2, 3, 4, 5], dtype='int64')
+
+        >>> kidx.putmask(kidx > 3, 100).sort_values()
+        Int64Index([1, 2, 3, 100, 100], dtype='int64')
+
+        >>> kidx.putmask(kidx > 3, ks.Index([100, 200, 300, 400, 500])).sort_values()
+        Int64Index([1, 2, 3, 400, 500], dtype='int64')
+        """
+        scol_name = self._internal.index_spark_column_names[0]
+        sdf = self._internal.spark_frame.select(self.spark.column)
+
+        dist_sequence_col_name = verify_temp_column_name(sdf, "__distributed_sequence_column__")
+        sdf = InternalFrame.attach_distributed_sequence_column(
+            sdf, column_name=dist_sequence_col_name
+        )
+
+        replace_col = verify_temp_column_name(sdf, "__replace_column__")
+        masking_col = verify_temp_column_name(sdf, "__masking_column__")
+
+        if isinstance(value, (list, tuple)):
+            replace_udf = udf(lambda x: value[x], _infer_type(value[0]))
```

Comment:
Is it possible to use `pandas_udf` instead of `udf`? If possible, could you replace with it?

Reply:
It is possible! I modified it to use `pandas_udf`